src/main/scala/net/psforever/actors/session/support/SessionData.scala

```diff
@@ -0,0 +1,3141 @@
+  val mountResponse: SessionMountHandlers =
+    new SessionMountHandlers(sessionData=this, avatarActor, context)
+  val terminals: SessionTerminalHandlers =
+    new SessionTerminalHandlers(sessionData=this, avatarActor, context)
+  private var _vehicleResponse: Option[SessionVehicleHandlers] = None
+  private var _galaxyResponse: Option[SessionGalaxyHandlers] = None
+  private var _squadResponse: Option[SessionSquadHandlers] = None
+  private var _zoning: Option[ZoningOperations] = None
+  def vehicleResponseOperations: SessionVehicleHandlers = _vehicleResponse.orNull
+  def galaxyResponseHanders: SessionGalaxyHandlers = _galaxyResponse.orNull
+  def squad: SessionSquadHandlers = _squadResponse.orNull
+  def zoning: ZoningOperations = _zoning.orNull
+
+  val sessionSupervisorStrategy: SupervisorStrategy = {
+    import net.psforever.objects.inventory.InventoryDisarrayException
+    OneForOneStrategy(maxNrOfRetries = -1, withinTimeRange = 1 minute) {
+      case nge: NoGUIDException =>
+        nge.getEntity match {
+          case p: Player =>
+            continent.GUID(p.VehicleSeated) match {
+              case Some(v: Vehicle) =>
+                attemptRecoveryFromNoGuidExceptionAsVehicle(v, nge)
+              case _ =>
+                attemptRecoveryFromNoGuidExceptionAsPlayer(p, nge)
+            }
+
+          case v: Vehicle =>
+            attemptRecoveryFromNoGuidExceptionAsVehicle(v, nge)
+
+          case e: Equipment =>
+            (
+              player.Holsters().zipWithIndex.flatMap { case (o, i) =>
+                o.Equipment match {
+                  case Some(e) => Some((player, InventoryItem(e, i)))
+                  case None => None
+                }
+              }.toList ++
+                player.Inventory.Items.map { o => (player, o) } ++ {
+                player.FreeHand.Equipment match {
+                  case Some(_) => List((player, InventoryItem(e, Player.FreeHandSlot)))
+                  case _ => Nil
+                }
+              } ++
+                (ValidObject(player.VehicleSeated) match {
+                  case Some(v: Vehicle) => v.Trunk.Items.map { o => (v, o) }
+                  case _ => Nil
+                })
+              )
+              .find { case (_, InventoryItem(o, _)) => o eq e } match {
+              case Some((c: Container, InventoryItem(obj, index))) =>
+                if (!obj.HasGUID) {
+                  c.Slot(index).Equipment = None
+                }
+                c match {
+                  case _: Player =>
+                    attemptRecoveryFromNoGuidExceptionAsPlayer(player, nge)
+                  case v: Vehicle =>
+                    if (v.PassengerInSeat(player).contains(0)) {
+                      attemptRecoveryFromNoGuidExceptionAsPlayer(player, nge)
+                    }
+                    SupervisorStrategy.resume
+                  case _ =>
+                    writeLogExceptionAndStop(nge)
+                }
```

Comment:
that can be more than one case `case ((v: Vehicule, ...)) => ...`

Reply:
Counterargument: if I break it down into it's three cases - `((c: Player, ...))`, `((c: Vehicle, ...))`, `((c: : Container, ...))` - then I have to replicate 176-178 three times.  I could also turn that into a guard boolean and break this down into __six__ case statements, half of which replicate the line `c.Slot(index).Equipment = None`.